lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -257,148 +266,262 @@ def common_problems(self, trans: ProvidesUserContext, id, **kwd):
-
-            Resolve parameters as a list for nested display. More client logic
-            here than is ideal but it is hard to reason about tool parameter
-            types on the client relative to the server. Job accessibility checks
-            are slightly different than dataset checks, so both methods are
-            available.
+        This method is designed to scan the list of previously run jobs and find records of jobs that had
+        the exact some input parameters and datasets. This can be used to minimize the amount of repeated work, and simply
+        recycle the old results.
+        """
+        tool_id = payload.tool_id
 
-            This API endpoint is unstable and tied heavily to Galaxy's JS client code,
-            this endpoint will change frequently.
+        tool = trans.app.toolbox.get_tool(tool_id)
+        if tool is None:
+            raise exceptions.ObjectNotFound("Requested tool not found")
+        # TODO the inputs are actually a dict, but are passed as a JSON dump
+        # maybe change it?
+        inputs = json.loads(payload.inputs)
+        # Find files coming in as multipart file data and add to inputs.
+        for k, v in payload.__annotations__.items():
+            if k.startswith("files_") or k.startswith("__files_"):
+                inputs[k] = v
+        request_context = WorkRequestContext(app=trans.app, user=trans.user, history=trans.history)
+        all_params, all_errors, _, _ = tool.expand_incoming(
+            trans=trans, incoming=inputs, request_context=request_context
+        )
+        if any(all_errors):
+            return []
+        params_dump = [tool.params_to_strings(param, trans.app, nested=True) for param in all_params]
+        jobs = []
+        for param_dump, param in zip(params_dump, all_params):
+            job = self.service.job_search.by_tool_input(
+                trans=trans,
+                tool_id=tool_id,
+                tool_version=tool.version,
+                param=param,
+                param_dump=param_dump,
+                job_state=payload.state,
+            )
+            if job:
+                jobs.append(job)
+        return [EncodedJobDetails(**single_job.to_dict("element")) for single_job in jobs]
 
-        :type   job_id: string
-        :param  job_id: Encoded job id
+    @router.get(
+        "/api/jobs/{id}",
+        name="show_job",
+        summary="Return dictionary containing description of job data.",
+    )
+    def show(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        full: Annotated[Optional[bool], FullShowQueryParam] = False,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> Dict[str, Any]:
+        return self.service.show(trans, id, bool(full))
 
-        :type   dataset_id: string
-        :param  dataset_id: Encoded HDA or LDDA id
+    # TODO find the mapping of the legacy route
+    # TODO rename? --> function cancels/stops job (no deletion involved?)
```

Comment:
```suggestion
```

methods that match the CRUD keywords are added automatically via https://github.com/mvdbeek/galaxy/blob/4e15e6929ef241d4d6428f84be14955f98c29cc4/lib/galaxy/webapps/galaxy/buildapp.py#L987 ... you should be able to delete this